engine/src/test/java/io/camunda/zeebe/engine/perf/TestEngine.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.perf;
+
+import io.camunda.zeebe.engine.processing.EngineProcessors;
+import io.camunda.zeebe.engine.processing.message.command.SubscriptionCommandSender;
+import io.camunda.zeebe.engine.processing.streamprocessor.JobStreamer;
+import io.camunda.zeebe.engine.state.DefaultZeebeDbFactory;
+import io.camunda.zeebe.engine.util.ProcessingExporterTransistor;
+import io.camunda.zeebe.engine.util.StreamProcessingComposite;
+import io.camunda.zeebe.engine.util.TestInterPartitionCommandSender;
+import io.camunda.zeebe.engine.util.TestStreams;
+import io.camunda.zeebe.engine.util.client.DeploymentClient;
+import io.camunda.zeebe.engine.util.client.ProcessInstanceClient;
+import io.camunda.zeebe.stream.impl.StreamProcessorMode;
+import io.camunda.zeebe.util.FeatureFlags;
+import java.util.ArrayList;
+import java.util.Optional;
+
+/** Helper class which should help to make it easy to create an engine for tests. */
+public final class TestEngine {
+
+  private final StreamProcessingComposite streamProcessingComposite;
+
+  private TestEngine(
+      final int partitionId, final int partitionCount, final TestContext testContext) {
+
+    final var testStreams =
+        new TestStreams(
+            testContext.temporaryFolder(),
+            testContext.autoCloseableRule(),
+            testContext.actorScheduler());
```

Comment:
:thought_balloon: Would be great to get rid of "rules" here. `TestStreams` really doesn't need an `AutoClosableRule` and the `TemporaryFolder` can be replaced by a simple `Path`.

Reply:
Yeah I wanted to stop at some point with the refactoring, I see this mostly in #10004